daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -726,6 +726,8 @@ private[lf] object Speedy {
 
     private val damlTraceLog = LoggerFactory.getLogger("daml.tracelog")
 
+    def newTraceLog: TraceLog = RingBufferTraceLog(damlTraceLog, 100)
```

Comment:
Where you interested in readability or were you unsure whether `RingBufferTraceLog` was actually constructed at every call? From some experimentation it looks like it should.

Reply:
neither really. I use this in `ScenarioRunner` and didn’t want to duplicate it.